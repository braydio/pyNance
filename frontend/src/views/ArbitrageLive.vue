<template>
  <BasePageLayout>
    <PageHeader :icon="Activity">
      <template #title>R/S Arbitrage Monitor</template>
      <template #subtitle>Live R/S arbitrage feed</template>
    </PageHeader>
    <pre
      v-if="content"
      class="bg-gray-800 text-white p-4 rounded whitespace-pre-wrap"
    >{{ content }}</pre>
    <div v-else>Loading...</div>
  </BasePageLayout>
</template>

<script setup>
import { ref, onMounted } from 'vue'
import { fetchArbitrageData } from '@/api/arbitrage'
import BasePageLayout from '@/components/layout/BasePageLayout.vue'
import PageHeader from '@/components/ui/PageHeader.vue'
import { Activity } from 'lucide-vue-next'

const content = ref('')

const load = async () => {
  try {
    const data = await fetchArbitrageData()
    content.value = data.content || JSON.stringify(data, null, 2)
  } catch (err) {
    console.error('Failed to load arbitrage data', err)
  }
}

onMounted(() => {
  load()
  setInterval(load, 5000)
})
</script>
