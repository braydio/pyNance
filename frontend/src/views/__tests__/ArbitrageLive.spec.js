// @vitest-environment jsdom
import { describe, it, expect, vi } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import ArbitrageLive from '../ArbitrageLive.vue'

vi.mock('@/api/arbitrage', () => ({
  fetchArbitrageData: vi.fn().mockResolvedValue({ content: '' })
}))

describe('ArbitrageLive.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(ArbitrageLive, {
      global: {
        stubs: ['BasePageLayout', 'PageHeader', 'Activity']
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
