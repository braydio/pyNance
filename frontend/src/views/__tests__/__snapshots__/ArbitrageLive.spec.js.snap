// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ArbitrageLive.vue > matches snapshot 1`] = `"<base-page-layout-stub padding="true" gap="gap-6"></base-page-layout-stub>"`;
